packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -153,9 +150,100 @@ export function generateTemplateMetadata(codeGen: CodeGen): t.Statement[] {
         metadataExpressions.push(t.expressionStatement(refsMetadata));
     }
 
+    const stylesheetsMetadata = t.assignmentExpression(
+        '=',
+        t.memberExpression(t.identifier(TEMPLATE_FUNCTION_NAME), t.identifier('stylesheets')),
+        t.arrayExpression([])
+    );
+    metadataExpressions.push(t.expressionStatement(stylesheetsMetadata));
+
+    const stylesheetTokens = generateStylesheetTokens(codeGen);
+    metadataExpressions.push(...stylesheetTokens);
+
+    const implicitStylesheetImports = generateImplicitStylesheetImports();
+    metadataExpressions.push(...implicitStylesheetImports);
+
     return metadataExpressions;
 }
 
+// Generates conditional statements to insert stylesheets into the
+// tmpl.stylesheets metadata.
+function generateImplicitStylesheetImports(): t.IfStatement[] {
+    // tmpl.stylesheets
+    const tmplStylesheetsExpr = t.memberExpression(
+        t.identifier(TEMPLATE_FUNCTION_NAME),
+        t.identifier('stylesheets')
+    );
+    // tmpl.stylesheets.push.apply
```

Comment:
We might as well avoid the `apply`; it is useless if we're already grabbing the `apply` off of the array.

Should be:

```js
tmpl.stylesheets.push(_implicitStylesheets);
```

Reply:
Actually let's do this in a follow-up PR; maybe it has some compat issues that will surprise us.